some cuda tests failing due to changes in error messages

<!--

Thanks for opening an issue! To help the Numba team handle your information
efficiently, please first ensure that there is no other issue present that
already describes the issue you have
(search at https://github.com/numba/numba/issues?&q=is%3Aissue).

For more general "how do I do X?" type questions, please speak to us in real
time on https://gitter.im/numba/numba or post to the Numba mailing list
https://groups.google.com/a/continuum.io/forum/#!forum/numba-users.

-->

## Reporting a bug

<!--

Before submitting a bug report please ensure that you can check off these boxes:

-->

- [x] I am using the latest released version of Numba (most recent is visible in
 the change log (https://github.com/numba/numba/blob/master/CHANGE_LOG).
- [x] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

<!--

Please include details of the bug here, including, if applicable, what you
expected to happen!

-->

```pytb
======================================================================
FAIL: test_global_build_list (numba.cuda.tests.cudapy.test_sm_creation.TestSharedMemoryCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/opt/numba/cuda/tests/cudapy/test_sm_creation.py", line 77, in test_global_build_list
str(raises.exception))
AssertionError: 'Invalid use of Function(<function shared.array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)\nNo implementation of function Function(<function shared.array at 0x7f527a848560>) found for signature:\n \n >>> array(shape=list(int64), dtype=class(float32))\n \nThere are 2 candidate implementations:\n  - Of which 2 did not match due to:\n  Overload in function \'array\': File: numba/cuda/stubs.py: Line 155.\n    With argument(s): \'(shape=list(int64), dtype=class(float32))\':\n   No match.\n\nDuring: resolving callee type: Function(<function shared.array at 0x7f527a848560>)\nDuring: typing of call at /opt/numba/cuda/tests/cudapy/test_sm_creation.py (27)\n\n\nFile "opt/conda/envs/testenv_4b89ae00-f772-429a-bef4-ad38c2966b5c/lib/python3.7/site-packages/numba/cuda/tests/cudapy/test_sm_creation.py", line 27:\ndef udt_global_build_list(A):\n    <source elided>\n    sa = cuda.shared.array(shape=[GLOBAL_CONSTANT, GLOBAL_CONSTANT_2],\n                           dtype=float32)\n                           ^\n'

======================================================================
FAIL: test_invalid_1 (numba.cuda.tests.cudapy.test_sm_creation.TestSharedMemoryCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/opt/numba/cuda/tests/cudapy/test_sm_creation.py", line 93, in test_invalid_1
str(raises.exception))
AssertionError: 'Invalid use of Function(<function shared.array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)\nNo implementation of function Function(<function shared.array at 0x7f527a848560>) found for signature:\n \n >>> array(shape=float32, dtype=class(float32))\n \nThere are 2 candidate implementations:\n    - Of which 2 did not match due to:\n    Overload in function \'array\': File: numba/cuda/stubs.py: Line 155.\n      With argument(s): \'(shape=float32, dtype=class(float32))\':\n     No match.\n\nDuring: resolving callee type: Function(<function shared.array at 0x7f527a848560>)\nDuring: typing of call at /opt/numba/cuda/tests/cudapy/test_sm_creation.py (39)\n\n\nFile "opt/conda/envs/testenv_4b89ae00-f772-429a-bef4-ad38c2966b5c/lib/python3.7/site-packages/numba/cuda/tests/cudapy/test_sm_creation.py", line 39:\ndef udt_invalid_1(A):\n    sa = cuda.shared.array(shape=A[0], dtype=float32)\n    ^\n'

======================================================================
FAIL: test_invalid_2 (numba.cuda.tests.cudapy.test_sm_creation.TestSharedMemoryCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/opt/numba/cuda/tests/cudapy/test_sm_creation.py", line 105, in test_invalid_2
str(raises.exception))
AssertionError: 'Invalid use of Function(<function shared.array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)\nNo implementation of function Function(<function shared.array at 0x7f527a848560>) found for signature:\n \n >>> array(shape=Tuple(Literal[int](1), array(float32, 1d, A)), dtype=class(float32))\n \nThere are 2 candidate implementations:\n      - Of which 2 did not match due to:\n      Overload in function \'array\': File: numba/cuda/stubs.py: Line 155.\n        With argument(s): \'(shape=Tuple(Literal[int](1), array(float32, 1d, A)), dtype=class(float32))\':\n       No match.\n\nDuring: resolving callee type: Function(<function shared.array at 0x7f527a848560>)\nDuring: typing of call at /opt/numba/cuda/tests/cudapy/test_sm_creation.py (45)\n\n\nFile "opt/conda/envs/testenv_4b89ae00-f772-429a-bef4-ad38c2966b5c/lib/python3.7/site-packages/numba/cuda/tests/cudapy/test_sm_creation.py", line 45:\ndef udt_invalid_2(A):\n    sa = cuda.shared.array(shape=(1, A[0]), dtype=float32)\n    ^\n'

======================================================================
FAIL: test_invalid_3 (numba.cuda.tests.cudapy.test_sm_creation.TestSharedMemoryCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/opt/numba/cuda/tests/cudapy/test_sm_creation.py", line 117, in test_invalid_3
str(raises.exception))
AssertionError: 'Invalid use of Function(<function shared.array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)\nNo implementation of function Function(<function shared.array at 0x7f527a848560>) found for signature:\n \n >>> array(shape=int32, dtype=class(float32))\n \nThere are 2 candidate implementations:\n      - Of which 2 did not match due to:\n      Overload in function \'array\': File: numba/cuda/stubs.py: Line 155.\n        With argument(s): \'(shape=int32, dtype=class(float32))\':\n       No match.\n\nDuring: resolving callee type: Function(<function shared.array at 0x7f527a848560>)\nDuring: typing of call at /opt/numba/cuda/tests/cudapy/test_sm_creation.py (39)\n\n\nFile "opt/conda/envs/testenv_4b89ae00-f772-429a-bef4-ad38c2966b5c/lib/python3.7/site-packages/numba/cuda/tests/cudapy/test_sm_creation.py", line 39:\ndef udt_invalid_1(A):\n    sa = cuda.shared.array(shape=A[0], dtype=float32)\n    ^\n'

======================================================================
FAIL: test_invalid_4 (numba.cuda.tests.cudapy.test_sm_creation.TestSharedMemoryCreation)
----------------------------------------------------------------------
Traceback (most recent call last):
File "/opt/numba/cuda/tests/cudapy/test_sm_creation.py", line 129, in test_invalid_4
str(raises.exception))
AssertionError: 'Invalid use of Function(<function shared.array' not found in 'Failed in nopython mode pipeline (step: nopython frontend)\nNo implementation of function Function(<function shared.array at 0x7f527a848560>) found for signature:\n \n >>> array(shape=Tuple(Literal[int](1), int32), dtype=class(float32))\n \nThere are 2 candidate implementations:\n      - Of which 2 did not match due to:\n      Overload in function \'array\': File: numba/cuda/stubs.py: Line 155.\n        With argument(s): \'(shape=Tuple(Literal[int](1), int32), dtype=class(float32))\':\n       No match.\n\nDuring: resolving callee type: Function(<function shared.array at 0x7f527a848560>)\nDuring: typing of call at /opt/numba/cuda/tests/cudapy/test_sm_creation.py (51)\n\n\nFile "opt/conda/envs/testenv_4b89ae00-f772-429a-bef4-ad38c2966b5c/lib/python3.7/site-packages/numba/cuda/tests/cudapy/test_sm_creation.py", line 51:\ndef udt_invalid_3(A):\n    sa = cuda.shared.array(shape=(1, A[0]), dtype=float32)\n    ^\n'

----------------------------------------------------------------------
```

--- a/numba/cuda/tests/cudapy/test_sm_creation.py
+++ b/numba/cuda/tests/cudapy/test_sm_creation.py
@@ -73,10 +73,11 @@ class TestSharedMemoryCreation(CUDATestCase):
         with self.assertRaises(TypingError) as raises:
             cuda.jit((float32[:, :],))(udt_global_build_list)
 
-        self.assertIn("Invalid use of Function(<function shared.array",
+        self.assertIn("No implementation of function "
+                      "Function(<function shared.array",
                       str(raises.exception))
-        self.assertIn("with argument(s) of type(s): "
-                      "(dtype=class(float32), shape=list(int64)",
+        self.assertIn("found for signature:\n \n "
+                      ">>> array(shape=list(int64), dtype=class(float32)",
                       str(raises.exception))
 
     def test_global_constant_tuple(self):
@@ -89,10 +90,11 @@ class TestSharedMemoryCreation(CUDATestCase):
         with self.assertRaises(TypingError) as raises:
             cuda.jit((float32[:],))(udt_invalid_1)
 
-        self.assertIn("Invalid use of Function(<function shared.array",
+        self.assertIn("No implementation of function "
+                      "Function(<function shared.array",
                       str(raises.exception))
-        self.assertIn("with argument(s) of type(s): "
-                      "(dtype=class(float32), shape=float32)",
+        self.assertIn("found for signature:\n \n "
+                      ">>> array(shape=float32, dtype=class(float32))",
                       str(raises.exception))
 
     @skip_on_cudasim("Can't check for constants in simulator")
@@ -101,10 +103,12 @@ class TestSharedMemoryCreation(CUDATestCase):
         with self.assertRaises(TypingError) as raises:
             cuda.jit((float32[:, :],))(udt_invalid_2)
 
-        self.assertIn("Invalid use of Function(<function shared.array",
+        self.assertIn("No implementation of function "
+                      "Function(<function shared.array",
                       str(raises.exception))
-        self.assertIn("with argument(s) of type(s): (dtype=class(float32), "
-                      "shape=Tuple(Literal[int](1), array(float32, 1d, A)))",
+        self.assertIn("found for signature:\n \n "
+                      ">>> array(shape=Tuple(Literal[int](1), "
+                      "array(float32, 1d, A)), dtype=class(float32))",
                       str(raises.exception))
 
     @skip_on_cudasim("Can't check for constants in simulator")
@@ -113,10 +117,11 @@ class TestSharedMemoryCreation(CUDATestCase):
         with self.assertRaises(TypingError) as raises:
             cuda.jit((int32[:],))(udt_invalid_1)
 
-        self.assertIn("Invalid use of Function(<function shared.array",
+        self.assertIn("No implementation of function "
+                      "Function(<function shared.array",
                       str(raises.exception))
-        self.assertIn("with argument(s) of type(s): "
-                      "(dtype=class(float32), shape=int32)",
+        self.assertIn("found for signature:\n \n "
+                      ">>> array(shape=int32, dtype=class(float32))",
                       str(raises.exception))
 
     @skip_on_cudasim("Can't check for constants in simulator")
@@ -125,10 +130,12 @@ class TestSharedMemoryCreation(CUDATestCase):
         with self.assertRaises(TypingError) as raises:
             cuda.jit((int32[:],))(udt_invalid_3)
 
-        self.assertIn("Invalid use of Function(<function shared.array",
+        self.assertIn("No implementation of function "
+                      "Function(<function shared.array",
                       str(raises.exception))
-        self.assertIn("with argument(s) of type(s): (dtype=class(float32), "
-                      "shape=Tuple(Literal[int](1), int32))",
+        self.assertIn("found for signature:\n \n "
+                      ">>> array(shape=Tuple(Literal[int](1), int32), "
+                      "dtype=class(float32))",
                       str(raises.exception))
 
 

